Products/CMFPlone/browser/password_reset.py

```diff
@@ -75,6 +84,40 @@ class PasswordResetView(BrowserView):
     form = ViewPageTemplateFile('templates/pwreset_form.pt')
     subpath = None
 
+    def _reset_password(self, pw_tool, randomstring):
+        userid = self.request.form.get('userid')
+        password = self.request.form.get('password')
+        try:
+            pw_tool.resetPassword(userid, randomstring, password)
+        except ExpiredRequestError:
+            return self.expired()
+        except InvalidRequestError:
+            return self.invalid()
+        except RuntimeError:
+            return self.invalid()
+        if True:  # TODO: get value from registry if we really want this
+            aclu = getToolByName(self.context, 'acl_users')
+            cups = aclu.plugins.listPlugins(ICredentialsUpdatePlugin)
+            for name, plugin in cups:
+                plugin.updateCredentials(
+                    self.request,
+                    self.request.response,
+                    userid,
+                    password
+                )
+            IStatusMessage(self.request).addStatusMessage(
+                _(
+                    'password_reset_successful',
+                    default='Password reset successful, '
+                            'you are logged in now!',
+                ),
+                'info',
+            )
+            url = INavigationRoot(self.context).absolute_url()
```

Comment:
I think we should use the login_next script so that the location is determined in the same way as for a normal login?

Reply:
This is designed to work with plone.login, so no such scripts...

Reply:
https://github.com/plone/Products.CMFPlone/issues/2439